website/docs/commands/0.13upgrade.html.markdown

```diff
@@ -0,0 +1,85 @@
+---
+layout: "docs"
+page_title: "Command: 0.13upgrade"
+sidebar_current: "docs-commands-013upgrade"
+description: |-
+  The 0.13upgrade subcommand updates existing configurations to use the new provider source features from Terraform 0.13.
+---
+
+# Command: 0.13upgrade
+
+The `terraform 0.13upgrade` command gathers provider source configuration to
```

Comment:
This opening sentence isn't very clear to me. The following is a rough suggestion of the information that ought to be conveyed in the first sentence, though my phrasing needs work:

The `terraform 0.13upgrade` command updates existing configuration to include a `source` attribute  in the `required_providers` setting in a `terraform` block for each provider required by the existing configuration.

Reply:
Thanks, that's helpful feedback! I tried to split your suggested sentence in two (aiming to communicate the most salient point first) and fiddled with it a bit. Here's my next pass:

> The `terraform 0.13upgrade` command updates existing configuration to add an
explicit `source` attribute for each provider used in a given module. The
provider source settings are stored in a `required_providers` block, by default
in the `versions.tf` file.

Any thoughts?

Reply:
very nice, that's much better! 

I might change the last sentence to this:

The
provider source settings are stored in a required_providers block, and 0.13upgrade puts those settings in a file named versions.tf.

It's _really_ nitpicky, so definitely don't let this block merging, but as written it seems to indicate that either:  terraform expects a file named versions.tf, or that the user has an option for a different output file name. if that's configurable - cool! ignore me!

Reply:
It's sort of configurable, in that if you have a single `required_providers` block already, we update that file instead of moving it to verisons.tf. But I'm finding that really hard to explain without distracting from the main point.

I'm finding it very difficult to be both concise and precise, but I don't think the level of precision in this sentence was useful. I've deleted the "by default in the `versions.tf` file" clause altogether and I think that's good enough for now. I added a note further down the docs that explains where the required providers configuration is stored in more detail.